ldap_auth_provider.py

```diff
@@ -69,9 +70,16 @@ def __init__(self, config: _LdapConfig, account_handler: ModuleApi):
 
         self.ldap_mode = config.mode
         self.ldap_uris = [config.uri] if isinstance(config.uri, str) else config.uri
-        self.ldap_tls = ldap3.Tls(
-            validate=ssl.CERT_REQUIRED if config.validate_cert else ssl.CERT_NONE
-        )
+        if config.tls_options:
+            if config.validate_cert:
+                logger.warning(
+                    "Config key validate_cert is not used when tls_options is specified"
+                )
```

Comment:
I think it would make more sense to move this logic into `parse_config`. I think it is probably clearer to either:

1. Error (we seem to just `raise Exception` in those cases) if both keys are set, or
2. Override `tls_options` with `validate_cert` if `validate_cert` is provided.

My preference, I think, would be the first one as it is clearer what is happening.

Reply:
Agreed!